plugins/plugin-github/codenvy-plugin-github-factory-resolver/src/main/java/com/codenvy/plugin/factory/github/factory/resolver/GithubFactoryParametersResolver.java

```diff
@@ -0,0 +1,177 @@
+ *
+ * @author Florent Benoit
+ */
+public class GithubFactoryParametersResolver implements FactoryParametersResolver {
+
+    /**
+     * Logger.
+     */
+    private static final Logger LOG = LoggerFactory.getLogger(GithubFactoryParametersResolver.class);
+
+    /**
+     * Parameter name.
+     */
+    protected static final String URL_PARAMETER_NAME = "url";
+
+    /**
+     * Connection timeout of 10seconds.
+     */
+    private static final int CONNECTION_TIMEOUT = (int) TimeUnit.SECONDS.toMillis(10);
+
+    /**
+     * Regexp to find repository details (repository name, project name and branch and subfolder)
+     * Examples of valid URLs are in the test class.
+     */
+    protected static final Pattern GITHUB_PATTERN = Pattern.compile("^(?:http)(?:s)?(?:\\:\\/\\/)github.com/(?<repoUser>[^/]++)/(?<repoName>[^/]++)(?:/tree/(?<branchName>[^/]++)(?:/(?<subFolder>.*))?)?$");
+
+    /**
+     * Check if this resolver can be used with the given parameters.
+     *
+     * @param factoryParameters
+     *         map of parameters dedicated to factories
+     * @return true if it will be accepted by the resolver implementation or false if it is not accepted
+     */
+    @Override
+    public boolean accept(@NotNull Map<String, String> factoryParameters) {
+        // Check if url parameter is a github URL
+        return factoryParameters.containsKey(URL_PARAMETER_NAME) && GITHUB_PATTERN.matcher(factoryParameters.get(URL_PARAMETER_NAME)).matches();
+    }
+
+    /**
+     * Create factory object based on provided parameters
+     *
+     * @param factoryParameters
+     *         map containing factory data parameters provided through URL
+     * @throws BadRequestException when data are invalid
+     */
+    @Override
+    public Factory createFactory(@NotNull Map<String, String> factoryParameters) throws BadRequestException {
+
+        // no need to check null value of url parameter as accept() method has performed the check
+        String githubUrl = factoryParameters.get("url");
+
+        // Apply github url to the regexp
+        Matcher matcher = GITHUB_PATTERN.matcher(githubUrl);
+        if (!matcher.matches()) {
+            throw new BadRequestException(String.format(
+                    "The given github url %s is not a valid URL github url. It should start with https://github.com/<user>/<repo>",
+                    githubUrl));
+        }
+
+        String repoUser = matcher.group("repoUser");
+        String repoName = matcher.group("repoName");
+        String branchName = matcher.group("branchName");
+        String subfolder = matcher.group("subFolder");
+
+        // build factory for the given github url
+        Map<String, String> parameters = new HashMap<>();
+        if (branchName == null) {
+            // use master if not specified
+            branchName = "master";
+        } else {
+            parameters.put("branch", branchName);
+        }
+        if (subfolder != null && !"/".equals(subfolder)) {
```

Comment:
Looks like subfolder group can be equal to `/` only if url ends with double slashes `https://github.com/user/repo/master//`.